src/backend/drm/atomic/surface.rs

```diff
@@ -39,14 +37,14 @@ pub struct Planes {
 }
 
 pub(in crate::backend::drm) struct AtomicDrmSurfaceInternal<A: AsRawFd + 'static> {
-    pub(super) dev: Rc<Dev<A>>,
+    pub(super) dev: Arc<Dev<A>>,
     pub(in crate::backend::drm) crtc: crtc::Handle,
-    pub(super) cursor: CursorState,
+    pub(super) cursor: Mutex<CursorState>,
     pub(in crate::backend::drm) planes: Planes,
     pub(super) state: RwLock<State>,
     pub(super) pending: RwLock<State>,
     pub(super) logger: ::slog::Logger,
-    pub(super) test_buffer: Cell<Option<(DumbBuffer, framebuffer::Handle)>>,
+    pub(super) test_buffer: Mutex<Option<(DumbBuffer, framebuffer::Handle)>>,
```

Comment:
Because our Surfaces need to be `Sync` (see below) all mutable access to surface internal data must be locked or be read-only.